synapse/handlers/auth.py

```diff
@@ -63,11 +63,11 @@ def __init__(self, hs):
         """
         super(AuthHandler, self).__init__(hs)
 
-        self.checkers = {}  # type: dict[str, UserInteractiveAuthChecker]
+        self.checkers = {}  # type: Dict[str, UserInteractiveAuthChecker]
         for auth_checker_class in INTERACTIVE_AUTH_CHECKERS:
             inst = auth_checker_class(hs)
             if inst.is_enabled():
-                self.checkers[inst.AUTH_TYPE] = inst
+                self.checkers[inst.AUTH_TYPE] = inst  # type: ignore
```

Comment:
I tried adding `AUTH_TYPE` as an abstract property to `UserInteractiveAuthChecker`, but then mypy complained about trying to instantiate an abstract class.